code/modules/projectiles/guns/specialist.dm

```diff
@@ -211,13 +212,14 @@ Note that this means that snipers will have a slowdown of 3, due to the scope
 
 
 /obj/item/weapon/gun/rifle/sniper/antimaterial/proc/laser_off(mob/user)
-	if(targetmarker_on)
-		if(laser_target)
-			deactivate_laser_target()
+	SIGNAL_HANDLER
+	if(laser_target)
+		deactivate_laser_target()
 		accuracy_mult -= 0.50 //We lose a big accuracy bonus vs the now unlasered target
 		STOP_PROCESSING(SSobj, src)
 		targetmarker_on = FALSE
 	targetmarker_primed = FALSE
+	UnregisterSignal(src, COMSIG_ITEM_UNZOOM)
```

Comment:
You're unregistering src from src anf registering from src to user?

Reply:
I am smrt